mutex/src/test/workflows.test.ts

```diff
@@ -0,0 +1,108 @@
+import { TestWorkflowEnvironment } from '@temporalio/testing';
+import { WorkflowHandle } from '@temporalio/client';
+import { Runtime, DefaultLogger, Worker } from '@temporalio/worker';
+import { describe, before, after, it } from 'mocha';
+import { currentWorkflowIdQuery, hasLockQuery, lockWorkflow, testLockWorkflow } from '../workflows';
+import assert from 'assert';
+import { v4 as uuidv4 } from 'uuid';
+
+const taskQueue = 'test' + new Date().toLocaleDateString('en-US');
+
+describe('lock workflow', function () {
+  let runPromise: Promise<void>;
+  let worker: Worker;
+  let lockWorkflowHandle: WorkflowHandle<typeof lockWorkflow>;
+  let env: TestWorkflowEnvironment;
+
+  before(async function () {
+    this.timeout(10000);
+    Runtime.install({ logger: new DefaultLogger('WARN') });
```

Comment:
No need to customize the logger anymore.
The default INFO level is fine IMHO, I changed the logger to use `process.stderr.write` directly instead of `console.error` and `jest` no longer warns about this.

I'd prefer to get the INFO messages and get a feel for what's going on even if the cost is more verbose output in tests.

Leaving it up to your consideration, run it and get a feel for it.

Reply:
I think it is better DX to suppress the additional log info in tests, at least for samples. Just having the output from the test runner helps new devs feel that nothing went wrong. The below output can be a bit jarring and make inexperienced users think an error occurred.

```
2023-01-05T15:50:06.637Z [INFO] Creating worker {
  options: {
    namespace: 'default',
    identity: '142091@defiant',
    shutdownGraceTime: '10s',
    maxConcurrentActivityTaskExecutions: 100,
    maxConcurrentLocalActivityExecutions: 100,
    enableNonLocalActivities: true,
    maxConcurrentWorkflowTaskExecutions: 100,
    stickyQueueScheduleToStartTimeout: '10s',
    maxHeartbeatThrottleInterval: '60s',
    defaultHeartbeatThrottleInterval: '30s',
    isolateExecutionTimeout: '5s',
    workflowThreadPoolSize: 8,
    maxCachedWorkflows: 261,
    enableSDKTracing: false,
    showStackTraceSources: false,
    debugMode: false,
    interceptors: { activityInbound: [Array], workflowModules: [Array] },
    sinks: { defaultWorkerLogger: [Object] },
    connection: NativeConnection {
      nativeClient: [External: 58c7630],
      referenceHolders: Set(0) {}
    },
    workflowsPath: '/home/val/Workspace/temporalio/samples-typescript/mutex/lib/workflows.js',
    activities: {
      useAPIThatCantBeCalledInParallel: [AsyncFunction: useAPIThatCantBeCalledInParallel]
    },
    taskQueue: 'test1/5/2023',
    shutdownGraceTimeMs: 10000,
    stickyQueueScheduleToStartTimeoutMs: 10000,
    isolateExecutionTimeoutMs: 5000,
    maxHeartbeatThrottleIntervalMs: 60000,
    defaultHeartbeatThrottleIntervalMs: 30000,
    loadedDataConverter: {
      payloadConverter: [DefaultPayloadConverter],
      failureConverter: [DefaultFailureConverter],
      payloadCodecs: []
    }
  }
}
2023-01-05T15:50:07.223Z [INFO] asset workflow-bundle-f8ba8cbc13e1ecef2505.js 703 KiB [emitted] [immutable] (name: main)
2023-01-05T15:50:07.224Z [INFO] modules by path ./node_modules/@temporalio/ 164 KiB
2023-01-05T15:50:07.224Z [INFO]   modules by path ./node_modules/@temporalio/common/lib/ 69 KiB
2023-01-05T15:50:07.224Z [INFO]     modules by path ./node_modules/@temporalio/common/lib/*.js 45 KiB 13 modules
2023-01-05T15:50:07.224Z [INFO]     modules by path ./node_modules/@temporalio/common/lib/converter/*.js 24 KiB 5 modules
2023-01-05T15:50:07.224Z [INFO]   modules by path ./node_modules/@temporalio/workflow/ 93.1 KiB
2023-01-05T15:50:07.224Z [INFO]     ./node_modules/@temporalio/workflow/lib/worker-interface.js 9.81 KiB [built] [code generated]
2023-01-05T15:50:07.224Z [INFO]     + 12 modules
2023-01-05T15:50:07.224Z [INFO]   ./node_modules/@temporalio/worker/lib/workflow-log-interceptor.js 2.27 KiB [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] modules by path ./lib/ 3.44 KiB
2023-01-05T15:50:07.224Z [INFO]   ./lib/workflows-autogenerated-entrypoint.cjs 535 bytes [built] [code generated]
2023-01-05T15:50:07.224Z [INFO]   ./lib/workflows.js 2.92 KiB [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] __temporal_custom_payload_converter (ignored) 15 bytes [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] __temporal_custom_failure_converter (ignored) 15 bytes [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] ./node_modules/long/umd/index.js 43.1 KiB [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] ./node_modules/ms/index.js 2.95 KiB [built] [code generated]
2023-01-05T15:50:07.224Z [INFO] webpack 5.75.0 compiled successfully in 412 ms
2023-01-05T15:50:07.226Z [INFO] Workflow bundle created { size: '0.69MB' }
2023-01-05T15:50:07.459909Z  INFO temporal_sdk_core::worker: Initializing worker task_queue=test1/5/2023 namespace=default
2023-01-05T15:50:07.460Z [INFO] Worker state changed { state: 'RUNNING' }

```

So I'd prefer to keep this. But happy to remove it if we want to remove this output across the board.

Reply:
I'm not religious about this, up to you.